9/README.md

```diff
@@ -0,0 +1,322 @@
+```
+
+* Values of DevVarLongStringArray and DevVarDoubleStringArray SHALL follow rules \<long-string-value\> 
+and \<double-string-value\>, respectively:
+```abnf
+ long-string-value = lvalue svalue 
+ double-string-value = dvalue svalue
+                       ; elements order implied by the above rules MAY be ignored
+```
+;where \<lvalue\> has type DevVarLongArray, \<dvalue\> has type DevVarDoubleArray and \<svalue\> has type 
+DevVarStringArray.
+
+#### State
+
+DavState type standardize a way of describing state of Device. Its values are from predefined set of 14 names.   
+
+* DevState name SHALL be according to the following rule:
+```abnf
+ DevState = "DevState
+```
+
+* Value of DevState type SHALL follow the rule \<dev-state-value\>:
+```abnf
+ dev-state-value = "ALARM" / "INSERT" / "STANDBY" / "CLOSE" / "MOVING" / "UNKNOWN" / "DISABLE" / "OFF"
+ dev-state-value =/ "EXTRACT" / "ON" / "FAULT" / "OPEN" / "INIT" / "RUNNING"
+```
+
+#### Enumeration
+
+DevEnum data type allows to assign string Labels to DevShort values. It is valid only for Attributes. 
+
+* DevEnum name SHALL follow the following rule:
+```abnf
+ DevEnum = "DevEnum"
+```
+
+* Label values of DevEnum SHALL follow the rule \<devenum-label\>:
+```abnf
+ devenum-label = 1*VCHAR
+```  
+
+* DevEnum labels SHALL be unique in the context of an Attribute.
+* Each of DevEnum labels SHALL correspond to one and unique DevShort value. 
+* Related DevShort values SHALL be consecutive and SHALL start with 0. 
+* Values of DevEnum DataType SHALL be transported as values of DevShort.
+
+* For any of Attribute of DevEnum data type, list of Labels SHALL be available as `enum_labels` Attribute property.
+
+* Tango Controls SHALL allow a Tango Client to retrieve labels associated with the DevShort value.
+
+#### Pipe
+
+DevPipeBlob is a Data Type to transfer data related to Pipes (see 7/Pipe).
+
+* DevPipeBlob SHALL have the following name:
+```abnf
+ DevPipeBlob = "DevPipeBlob"
+```
+
+* A value of DevPipeBlob type SHALL follow the \<pipe-blob-value\> rule:
+
+```abnf
+ pipe-blob-value = name blob-data
+                   ; elements order implied by the above rule MAY be ignored
+
+ name = 1*VCHAR 
+
+ blob-data = *blob-data-element
+ 
+ blob-data-element = name value inner-blob inner-blob-name 
+                     ; elements order implied by the above rule MAY be ignored    
+
+ inner-blob = blob-data  
+```
+
+#### Exceptions 
+
+Results of failed operations (exceptions) within the Tango Controls are send as values/messages of DevFailed data type.
+
+* DevFailed type name is as follows:
+```abnf
+ DevFailed = "DevFailed"
+```
+
+* Any value of DevFiled type SHALL follow \<dev-failed-value\> rule:
+```abnf
+ dev-failed-value = 1*error 
+
+ error = reason severity desc origin
+
+ reason = *VCHAR ; SHALL be the reason of the failur
+ severity = "WARN" | "ERR" | "PANIC"
+ desc = *VCHAR ; SHOULD describe the failur in more details, eg. call stack
```

Comment:
```suggestion
 desc = *VCHAR ; SHOULD describe the failure in more details
```
For me:
-  the call stack is in the "origin" field and not the "desc" field.
- desc is a message describing what is occuring in english.
- "reason" is string without spaces that is standardized for some errors thrown by the API. examples: "API_AttrOptProp", "API_AttrNotPolled","API_CmdNotPolled", "API_CommandNotFound", .... 
@bourtemb What do you think? Should we list all reasons of the API in the Tango RFC? I have found 39 of them in the Java code.

Reply:
@gwen-soleil, I think it's good to standardize the reason field for the most common errors, and for this I think it's good to have a place where these errors are listed, but I don't think this RFC is the best location to list them all. Maybe we can add a comment as you did saying that the reason field is standardized for some errors thrown by the API.
We would probably need another RFC (_Errors_ or something like that) to list all of these standard errors.

Reply:
I have checked the tango_const.h and there is about 130 definitions of API_... reasons. However, there is no mechanism for forcing usage of only these names for reasons.  Are these compatible between java and c++ implementations?

Reply:
I suggest changing SHALL to SHOULD, and skip listing all the possible reasons because if there will be a need to add a new one in the RFC will be invalidated and possibly not compatible between java and c++.

Reply:
> I have checked the tango_const.h and there is about 130 definitions of API_... reasons. However, there is no mechanism for forcing usage of only these names for reasons. Are these compatible between java and c++ implementations?

I guess many of them are compatible because the Java API was developed after the C++ one and was mainly based on it at the beginning so I guess many errors are common in these 2 languages but I would not be surprised to see some differences.
This is why I find it interesting to have another RFC listing at least the standardized reason fields for the most common errors coming from a Tango implementation, with a description of these errors if it makes sense.
Then, 2 different Tango implementations could refer to this RFC and use consistent reason fields for similar errors.

Reply:
You have a point. I will add this RFC to the list of ones which are waiting to be added :)